### PR TITLE
Add changelog v1.16.1

### PR DESCRIPTION
**Pull request by Hoxbro**
Needs: 
- [x] https://github.com/holoviz/holoviews/pull/5733
- [x] https://github.com/holoviz/holoviews/pull/5729
- [x] Update releases.rst

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,29 @@
+Version 1.16.1
+==============
+**June 2, 2023**
+
+This release contains a small number of enhancements and important bug fixes. Many thanks to @philippjfr
```
Comment (jlstevens) [inline CHANGELOG.md:5]: I would swap the order round, first mention the important bug fixes then the small enhancements...

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,29 @@
...
+and @Hoxbro.
+
+This release includes a breaking change as notebooks will no longer be inlining as default. This change will reduce the size of the notebook files and, more importantly, address an upstream issue where running `hv.extension` would give an error when used in a notebook.
```
Comment (jlstevens) [inline CHANGELOG.md:8]: I would consider 'upstream issue in Jupyter' to be a little more specific...

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,29 @@
...
+
+
+Enhancements:
```
Comment (jlstevens) [inline CHANGELOG.md:11]: I would put the section where you describe the fix for `inline=True` before the enhancements section.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,29 @@
...
+
+- Add `show_versions` helper function ([#5725](https://github.com/holoviz/holoviews/pull/5725))
+- Ignore known model warning in VS Code ([#5734](https://github.com/holoviz/holoviews/pull/5734))
+- Add work-around for plots where the objects in figure would overlap ([#5733](https://github.com/holoviz/holoviews/pull/5733))
+
+Compatibility:
```
Comment (jlstevens) [inline CHANGELOG.md:17]: 'Compatibility and critical bug fixes'?

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,30 @@
...
+
+This release contains a small number of important bug fixes and enhancements. Many thanks to @philippjfr
+and @Hoxbro.
+
+This release includes a breaking change as notebooks will no longer be inlining as default. This change will reduce the size of the notebook files and, more importantly, address an upstream issue in Jupyter where running `hv.extension` would give an error when used in a notebook.
+
+
+Critical bug fixes and compatibility:
+
+- Add `enable_mathjax` and set it and inline to `False` ([#5733](https://github.com/holoviz/holoviews/pull/5733))
```
[the remark this answers is not in the document]
Comment (jlstevens) [reply CHANGELOG.md:12]: Also, this PR number seems wrong? I get 'Add workaround for numpy arrays with object dtype'...

[the remark this answers is not in the document]
Comment (Hoxbro) [reply CHANGELOG.md]: Thank you :+1:

**Pull request merged**